### ISSUE TITLE
nginx-auth does not work on ubuntu 16.04 (xenial)

### ISSUE THREAD

**Issue by mhlakhani** [tailscale/tailscale#4817]
### What is the issue?

I tried following the instructions in [this blog post](https://tailscale.com/blog/tailscale-auth-nginx/) but was unable to get them to work.

After some debugging it looks like the systemd service refuses to start with an error saying "unknown lvalue 'DynamicUser' in section 'Service'", presumably pointing to this line: https://github.com/tailscale/tailscale/blob/main/cmd/nginx-auth/tailscale.nginx-auth.service#L8

From some searching I believe that feature was added in [systemd 235](https://0pointer.net/blog/dynamic-users-with-systemd.html), while Xenial ships older versions (229 I believe, [per this link](https://launchpad.net/ubuntu/xenial/+source/systemd))

Should I consider it unsupported on ubuntu 16.04, or is there a workaround? I was assuming this would be supported as the package is available in the tailscale repos for Xenial.

I'm not blocked on this anymore as I upgraded my system, but still thought this was worth flagging.

### OS

Linux

### OS version

Ubuntu 16.04 Xenial

### Tailscale version

1.24.2

### Bug report

BUG-fda7735932997695d0a5d63c0a7dc098a98e9237d6736d1d48268c6d514060a3-20220608015629Z-3da2814c2214800d

**Comment by danderson** [comment]
Ubuntu 16.04 hit EOL a year ago, so is no longer maintained unless you're paying for extended security updates. I wouldn't recommend running stuff on a platform that's not getting security fixes, in general.

That said, as a workaround, removing the `DynamicUser` line from the systemd unit should let the proxy start up correctly. However, it'll be running as root with no permissions dropped, so depending on your comfort level that might be a bit scary. You can substitute `User` and `Group` directives in the place where `DynamicUser` was, to make the auth service run as an unprivileged user of your choice, which is closer to how `DynamicUser` runs the service.

**Comment by danderson** [comment]
Oh, duh, I missed the line where you said you already upgraded away from 16.04 *facepalm* sorry, been a long day.

So yeah, our usual line for things like this is that we don't support OSes that are themselves unsupported. We don't go out of our way to break them, but in this case `DynamicUser` is a 4+ year old feature, so it feels fair game to use here.

It might still be worth tweaking, in case ancient but still-supported RHELs or somesuch still use a very old systemd, but I don't know where the line is exactly. @Xe up to you as the author of the proxy, I think.

**Comment by mhlakhani** [comment]
Thanks for the quick response @danderson!

I mostly filed this as my experience so far with tailscale has been that everything works right out of the box so I figured that if this is a published package for my system then it should work (tailscale itself worked fine).

that said, it makes a lot of sense to not support really old systems so I’m fine with this being a wontfix and defer to you folks.

thanks again for a great piece of software!

**Comment by DentonGentry** [comment]
To my knowledge there is not an alternate way for nginx-auth to work if DynamicUser is not available.

**Comment by DentonGentry** [comment]
I don't think we have a way to support Ubuntu 16.04 for this.
